tests/unit_tests/restruct_poc/test_cantilever_beam.cpp

```diff
@@ -99,8 +99,18 @@ TEST(DynamicBeamTest, CantileverBeamSineLoad) {
     const double rho_inf(0.0);
 
     // Create solver
+    auto nodes_vector = std::vector<Node>{};
+    for (const auto& node : model.GetNodes()) {
+        nodes_vector.push_back(*node);
+    }
+
+    auto constraints_vector = std::vector<Constraint>{};
+    for (const auto& constraint : model.GetConstraints()) {
+        constraints_vector.push_back(*constraint);
+    }
+
```

Comment:
Could the solver constructor be changed to take what is returned by `model.GetConstraints` and `model.GetNodes` copy doesn't need to be created manually? Either that or have `model.GetConstraints` return `std::vector<Constraint>`. I think the former would be easier to understand though.

Reply:
I looked into implementing this and this appears to be little more involved than what I initially thought. Let's refactor the `Solver` class in a follow up work.

Reply:
Never mind, I actually implemented a fix for the above in this PR.